source/agora/consensus/EnrollmentManager.d

```diff
@@ -423,6 +423,35 @@ public class EnrollmentManager
         return false;
     }
 
+    /***************************************************************************
+
+        Get the unregistered enrollments in the block
+
+        Params:
+            enrolls = will contain the unregistered enrollments data if found
+
+        Returns:
+            Return true if the unregistered enrollment data was found
+
+    ***************************************************************************/
+
+    public bool getEnrollments (out Enrollment[] enrolls) @trusted
+    {
+        enrolls.length = 0;
```

Comment:
If your goal here is to reuse memory, you should use `assumeSafeAppend`

Reply:
I add `assumeSafeAppend(enrolls)`